lib/IRGen/GenProto.cpp

```diff
@@ -1276,21 +1279,22 @@ class AccessorConformanceInfo : public ConformanceInfo {
     WitnessTableBuilder(IRGenModule &IGM, ConstantArrayBuilder &table,
                         SILWitnessTable *SILWT)
         : IGM(IGM), Table(table), SILWT(SILWT),
-          ConcreteType(SILWT->getConformance()->getDeclContext()
-                          ->mapTypeIntoContext(
-                            SILWT->getConformance()->getType())
-                         ->getCanonicalType()),
+          ConcreteType(
+              SILWT->getConformance()
+                  ->getDeclContext()
+                  ->mapTypeIntoContext(SILWT->getConformance()->getType())
+                  ->getCanonicalType()),
           Conformance(*SILWT->getConformance()),
-          ConformanceInContext(
-            mapConformanceIntoContext(IGM, Conformance,
-                                      Conformance.getDeclContext())),
+          ConformanceInContext(mapConformanceIntoContext(
+              IGM, Conformance, Conformance.getDeclContext())),
           SILEntries(SILWT->getEntries()),
           SILConditionalConformances(SILWT->getConditionalConformances()),
-          ResilientConformance(isResilientConformance(&Conformance)),
+          ResilientConformance(
+              isResilientConformance(IGM.Triple, &Conformance)),
           PI(IGM.getProtocolInfo(SILWT->getConformance()->getProtocol(),
                                  (ResilientConformance
-                                  ? ProtocolInfoKind::RequirementSignature
-                                  : ProtocolInfoKind::Full))) {
+                                      ? ProtocolInfoKind::RequirementSignature
+                                      : ProtocolInfoKind::Full))) {
       // If the conformance is resilient, we require runtime instantiation.
```

Comment:
I don't think you need any of the other changes in this PR. Instead, just add a check for the triple here and set ResilientConformance to true.

Also just as with class instantiation I would prefer it if you added a new flag to IRGenOptions instead of checking the target here directly.

Reply:
Hmm, okay, that does simplify things.  Unfortunately, when building the standard library itself is where things get interesting, and what I'm trying to figure out still.  It seems that `getFunctionIndex` is asserting at `getKind() >= ProtocolInfoKind::Full` since we are now just doing the `RequirementSignature`

Reply:
Sorry, I meant ‘or’ the additional condition when you set the ResilientConformance flag and not when you use it.